tests/regression_functional_test.go

```diff
@@ -0,0 +1,122 @@
+package tests_test
+
+import (
+	"flag"
+	"github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"kubevirt.io/kubevirt-ansible/tests"
+	"time"
+)
+
+//// The list of supported Operating Systems
+//// name : path_to_yml
+var osDict = map[string]string{
+	"testvm-fedora" : "manifests/vm-template-fedora.yml",
+}
+
+const (
+	temporaryJson = "/tmp/tmp-vm.yml"
+	ipQuery = "-o=jsonpath='{.status.interfaces[0].ipAddress}'"
+	phaseQuery = "-o=jsonpath='{.status.phase}'"
+)
+
+
+var _ = Describe("Regression and Functional tests of VMs and VMIs", func() {
+
+	flag.Parse()
+
+	Describe("Creating VMs with different OS from templates via oc and kubectl", func() {
+
+		// Run tests for each OS:
+		for k := range osDict {
+
+			osName := k
+			templatePath := osDict[k]
+
+			Context("Create/Delete " + osName + " VM from template via oc command", func() {
+
+				It("Create " + osName + " VM, start and check status", func() {
+					By("Create VM")
+					tests.ProcessTemplateWithParameters(templatePath, temporaryJson, "NAME="+osName, "CPU_CORES=2")
+					tests.CreateResourceWithFilePathTestNamespace(temporaryJson)
+
+					By("Start VM")
+					_, _, err := tests.StartStopVirtualMachineVirtctl(osName, "start")
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Waiting until VMI is running")
+					tests.WaitUntilResourceReadyByNameTestNamespace("vmi", osName, phaseQuery, "Running")
+
+					ipAddress := tests.GetVirtualMachineSpecificParameters("vmi", osName, ipQuery)
+					// removing quotes
+					ipAddress = ipAddress[1 : len(ipAddress)-1]
+
+					By("Waiting for the console and check IP address (now only for Fedora)")
+					if osName == "testvm-fedora" {
+						expecter, err := tests.LoggedInFedoraExpecter(osName, tests.NamespaceTestDefault, 300)
```

Comment:
Please create generic exepctor that will handle all images (can use switch case https://tour.golang.org/flowcontrol/9)

Reply:
We are not using the generic expecter, because it would be too complicated to write one. Each VM requires different steps to get log in. This approach is fine.

Reply:
changed to using "switch". If OS is not supported - console would not be checked.